ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3PostHandler.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+package com.github.ambry.frontend.s3;
+
+import com.github.ambry.commons.Callback;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.router.ReadableStreamChannel;
+
+
+/**
+ * Handles S3 POST requests.
+ */
+public class S3PostHandler {
+  private static final String UPLOADS_PARAM_NAME = "uploads";
+  private static final String UPLOAD_ID_PARAM_NAME = "uploadId";
+  private final S3MultipartPostHandler multipartPostHandler;
+
+  /**
+   * Constructs a handler for uploading s3 POST requests.
+   */
+  public S3PostHandler(S3MultipartPostHandler multipartPostHandler) {
+    this.multipartPostHandler = multipartPostHandler;
+  }
+
+  /**
+   * Handles a request for putting a blob.
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   */
+  public void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      Callback<ReadableStreamChannel> callback) {
+    if (isMultipartUploadRequest(restRequest)) {
+      multipartPostHandler.handle(restRequest, restResponseChannel, callback);
+    }
+  }
+
+  private boolean isMultipartUploadRequest(RestRequest restRequest) {
+    return restRequest.getArgs().containsKey(UPLOADS_PARAM_NAME) || restRequest.getArgs()
```

Comment:
According to the [spec](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html), the request should only contain uploadId, and it's mandatory

Reply:
Hmm, looks like S3 `CreateMultipartUpload` requests would contain `uploads` and `UploadPart` would contain `uploadId` as params.

https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html